Source-build OpenSSL linker errors

Source build is trying to build preview8 and we're running into linker errors in coreclr installer with what looks like OpenSSL:

RHEL7
```
  /home/chris/dotnet_source-build/artifacts/src/runtime.bf456654f9a4f9a86c15d9d50095ff29cde5f0a4/artifacts/bin/native/net5.0-Linux-Release-x64/libSystem.Security.Cryptography.Native.OpenSsl.a(apibridge.c.o): In function `local_X509_get_version':
  /home/chris/dotnet_source-build/artifacts/src/runtime.bf456654f9a4f9a86c15d9d50095ff29cde5f0a4/src/libraries/Native/Unix/System.Security.Cryptography.Native/apibridge.c:65: undefined reference to `ASN1_INTEGER_get'
  /home/chris/dotnet_source-build/artifacts/src/runtime.bf456654f9a4f9a86c15d9d50095ff29cde5f0a4/artifacts/bin/native/net5.0-Linux-Release-x64/libSystem.Security.Cryptography.Native.OpenSsl.a(apibridge.c.o): In function `local_OpenSSL_version_num':
  /home/chris/dotnet_source-build/artifacts/src/runtime.bf456654f9a4f9a86c15d9d50095ff29cde5f0a4/src/libraries/Native/Unix/System.Security.Cryptography.Native/apibridge.c:146: undefined reference to `SSLeay'
  /home/chris/dotnet_source-build/artifacts/src/runtime.bf456654f9a4f9a86c15d9d50095ff29cde5f0a4/artifacts/bin/native/net5.0-Linux-Release-x64/libSystem.Security.Cryptography.Native.OpenSsl.a(apibridge.c.o): In function `local_DSA_set0_pqg':
  /home/chris/dotnet_source-build/artifacts/src/runtime.bf456654f9a4f9a86c15d9d50095ff29cde5f0a4/src/libraries/Native/Unix/System.Security.Cryptography.Native/apibridge.c:214: undefined reference to `BN_free'
  /home/chris/dotnet_source-build/artifacts/src/runtime.bf456654f9a4f9a86c15d9d50095ff29cde5f0a4/src/libraries/Native/Unix/System.Security.Cryptography.Native/apibridge.c:220: undefined reference to `BN_free'
  /home/chris/dotnet_source-build/artifacts/src/runtime.bf456654f9a4f9a86c15d9d50095ff29cde5f0a4/src/libraries/Native/Unix/System.Security.Cryptography.Native/apibridge.c:226: undefined reference to `BN_free'
```

RHEL8: 
```
  /usr/bin/ld: /mnt/sdc/chris/dotnet_source-build/artifacts/src/runtime.bf456654f9a4f9a86c15d9d50095ff29cde5f0a4/artifacts/bin/native/net5.0-Linux-Release-x64/libSystem.Security.Cryptography.Native.OpenSsl.a(pal_evp_cipher.c.o): undefined reference to symbol 'EVP_des_ecb@@OPENSSL_1_1_0'
  //usr/lib64/libcrypto.so.1.1: error adding symbols: DSO missing from command line
```

Full logs are attached.  It seems like this might be related to https://github.com/dotnet/runtime/pull/39490/files.  Any ideas what could be going on?  cc @jeffschwMSFT @VSadov @safern @ViktorHofer @vitek-karas @swaroop-sridhar @agocke @omajid 
[rhel7-Linux.x64.Release-CMakeError.log](https://github.com/dotnet/runtime/files/5165060/rhel7-Linux.x64.Release-CMakeError.log)
[rhel7-Linux.x64.Release-CMakeOutput.log](https://github.com/dotnet/runtime/files/5165061/rhel7-Linux.x64.Release-CMakeOutput.log)
[rhel7-net5.0-Linux-Release-x64-CMakeError.log](https://github.com/dotnet/runtime/files/5165062/rhel7-net5.0-Linux-Release-x64-CMakeError.log)
[rhel7-net5.0-Linux-Release-x64-CMakeOutput.log](https://github.com/dotnet/runtime/files/5165063/rhel7-net5.0-Linux-Release-x64-CMakeOutput.log)
[rhel7-runtime.log](https://github.com/dotnet/runtime/files/5165064/rhel7-runtime.log)
[rhel.7-x64.Release-CMakeError.log](https://github.com/dotnet/runtime/files/5165065/rhel.7-x64.Release-CMakeError.log)
[rhel.7-x64.Release-CMakeOutput.log](https://github.com/dotnet/runtime/files/5165066/rhel.7-x64.Release-CMakeOutput.log)
[rhel8-net5.0-Linux-Release-x64-CMakeError.log](https://github.com/dotnet/runtime/files/5165074/rhel8-net5.0-Linux-Release-x64-CMakeError.log)
[rhel8-net5.0-Linux-Release-x64-CMakeOutput.log](https://github.com/dotnet/runtime/files/5165076/rhel8-net5.0-Linux-Release-x64-CMakeOutput.log)
[rhel8-runtime.log](https://github.com/dotnet/runtime/files/5165077/rhel8-runtime.log)
[rhel.8-x64.Release-CMakeError.log](https://github.com/dotnet/runtime/files/5165078/rhel.8-x64.Release-CMakeError.log)
[rhel.8-x64.Release-CMakeOutput.log](https://github.com/dotnet/runtime/files/5165079/rhel.8-x64.Release-CMakeOutput.log)
[rhel8-Linux.x64.Release-CMakeError.log](https://github.com/dotnet/runtime/files/5165080/rhel8-Linux.x64.Release-CMakeError.log)
[rhel8-Linux.x64.Release-CMakeOutput.log](https://github.com/dotnet/runtime/files/5165081/rhel8-Linux.x64.Release-CMakeOutput.log)

--- a/src/installer/corehost/build.sh
+++ b/src/installer/corehost/build.sh
@@ -92,6 +92,7 @@ export __BinDir __IntermediatesDir __CoreClrArtifacts __NativeLibsArtifacts
 __CMakeArgs="-DCLI_CMAKE_HOST_VER=\"$__host_ver\" -DCLI_CMAKE_COMMON_HOST_VER=\"$__apphost_ver\" -DCLI_CMAKE_HOST_FXR_VER=\"$__fxr_ver\" $__CMakeArgs"
 __CMakeArgs="-DCLI_CMAKE_HOST_POLICY_VER=\"$__policy_ver\" -DCLI_CMAKE_PKG_RID=\"$__DistroRid\" -DCLI_CMAKE_COMMIT_HASH=\"$__commit_hash\" $__CMakeArgs"
 __CMakeArgs="-DCORECLR_ARTIFACTS=\"$__CoreClrArtifacts\" -DNATIVE_LIBS_ARTIFACTS=\"$__NativeLibsArtifacts\" $__CMakeArgs"
+__CMakeArgs="-DFEATURE_DISTRO_AGNOSTIC_SSL=$__PortableBuild $__CMakeArgs"
 
 if [[ "$__PortableBuild" == 1 ]]; then
     __CMakeArgs="-DCLI_CMAKE_PORTABLE_BUILD=1 $__CMakeArgs"

--- a/src/installer/corehost/cli/apphost/static/CMakeLists.txt
+++ b/src/installer/corehost/cli/apphost/static/CMakeLists.txt
@@ -191,6 +191,11 @@ else()
     #     )
     endif ()
 
+    # Additional requirements for System.Security.Cryptography.Native.OpenSsl
+    if(NOT FEATURE_DISTRO_AGNOSTIC_SSL)
+        find_package(OpenSSL)
+    endif()
+
     if(CLR_CMAKE_TARGET_LINUX OR CLR_CMAKE_TARGET_FREEBSD OR CLR_CMAKE_TARGET_NETBSD OR CLR_CMAKE_TARGET_SUNOS)
         # These options are used to force every object to be included even if it's unused.
         set(START_WHOLE_ARCHIVE -Wl,--whole-archive)
@@ -213,3 +218,10 @@ target_link_libraries(singlefilehost
     ${NATIVE_LIBS_EXTRA}
 
 )
+
+if(NOT FEATURE_DISTRO_AGNOSTIC_SSL)
+    target_link_libraries(singlefilehost
+        ${OPENSSL_CRYPTO_LIBRARY}
+        ${OPENSSL_SSL_LIBRARY}
+    )
+endif()

